store/src/main/java/com/dropbox/android/external/store4/StoreBuilder.kt

```diff
@@ -54,6 +54,8 @@ interface StoreBuilder<Key : Any, Output : Any> {
      */
     fun disableCache(): StoreBuilder<Key, Output>
 
+    fun<UnwrappedOutput> fetcherTransformer(fetcherTransformer: (Output) -> FetcherResult<UnwrappedOutput>)
```

Comment:
Ignore anything in this file aside from this line. Everything bellow is a mess that I didn't want to get into before we decide this is the way to go.

I think the only scalable way to add support to another generic type changing parameter (like source of truth) would be for change the builder API it only receive the source of truth in the constructor.

Aside from builder changes (and tests) all the other changes needed to support this change are in the PR (and are pretty minimal)